connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufSerializer.scala

```diff
@@ -110,6 +110,19 @@ private[sql] class ProtobufSerializer(
               enumSymbols.mkString("\"", "\", \"", "\""))
           }
           fieldDescriptor.getEnumType.findValueByName(data)
+      case (IntegerType, ENUM) =>
+        val enumValues: Set[Int] =
+          fieldDescriptor.getEnumType.getValues.asScala.map(e => e.getNumber).toSet
+        (getter, ordinal) =>
+          val data = getter.getInt(ordinal)
+          if (!enumValues.contains(data)) {
+            throw QueryCompilationErrors.cannotConvertCatalystTypeToProtobufEnumTypeError(
```

Comment:
This is not type conversion error. Also it occurs at runtime, not at compile time. Check for other errors to see there is a better suited one.

Reply:
yeah i agree, i copy/pasted the error from the existing logic above.

i've moved the error from `QueryCompilationErrors` to `QueryExecutionErrors` and slightly renamed it, wdyt? happy to change more or use a different error if that makes more sense though